src/viam/logging.py

```diff
@@ -29,6 +29,12 @@ def __init__(self, parent: "RobotClient"):
         addHandlers(self._logger, True)
         super().__init__()
         self._logger.setLevel(self.level)
+        try:
+            self.loop = asyncio.get_event_loop()
+        except RuntimeError:
+            # If the log is coming from a thread that doesn't have an event loop, create and set a new one.
+            self.loop = asyncio.new_event_loop()
```

Comment:
[nit] we should probably at least log a warning that we are doing this? Could lead to weird behavior since we are basically expecting the user to write non-blocking async code in their thread.